drivers/scheduler/k8s/k8s.go

```diff
@@ -897,6 +890,12 @@ func (k *K8s) createCoreObject(spec interface{}, ns *v1.Namespace, app *spec.App
 	if obj, ok := spec.(*appsapi.Deployment); ok {
 		obj.Namespace = ns.Name
 		obj.Spec.Template.Spec.Volumes = k.substituteNamespaceInVolumes(obj.Spec.Template.Spec.Volumes, ns.Name)
+		if isDeploymentAppEnvEnabled(obj.Annotations) && len(options.AppEnvs) != 0 {
```

Comment:
Do we need the `isDeploymentAppEnvEnabled`? Why not always append the env variables if they are defined in the options.

Trying to keep the test logic simple with lesser if checks. That would also be simpler for the user of Schedule. Just add the env options during the call and no need to make any annotation changes to the yamls.